src/h5web/providers/Provider.tsx

```diff
@@ -9,15 +10,18 @@ interface Props {
 function Provider(props: Props): JSX.Element {
   const { api, children } = props;
 
-  if (!api) {
+  // Wait until metadata is fetched before rendering app
+  const { value: metadata } = useAsync(async () => api?.getMetadata(), [api]);
```

Comment:
The `useAsync` is to trigger a render at the promise resolution ?

Reply:
It's like a `useState` combined with a `useEffect` for making requests and storing the result. It also manages loading/error state.